### PR TITLE
Static HLE implementation

### PR DESCRIPTION
**Pull request by RipleyTom**
This is a basic implementation of static function hooking.
The functions implemented are:
-memcpy
-memset
-memmove
-memcmp
and the wchar_t variants of those functions(wmemcpy, etc).

To be active the hook static functions has to be checked in debug tab.

I've not noticed any performance gain from it on the few games I've tested it on but since it's done, here it is.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/StaticHLE.cpp [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+}
+
+bool StaticHleHandler::CheckAgainstPatterns(vm::cptr<u8>& data, u32 size, u32 addr)
+{
+	for (auto& pat : hle_patterns)
+	{
+		if (size < pat.total_length)
+			continue;
+
+		//	check start pattern
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/StaticHLE.cpp]: The comments in this file should be seperated by one space from the //

File: rpcs3/Emu/Cell/PPUModule.cpp [hunk 2]
```diff
@@ -1135,6 +1138,19 @@ void ppu_load_exec(const ppu_exec_object& elf)
 	// Initialize HLE modules
 	ppu_initialize_modules(link);
 
+	//Static HLE patching
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/PPUModule.cpp]: This comment should be seperated by one space from the //

File: rpcs3/Emu/Cell/PPUModule.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#include "stdafx.h"
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/PPUModule.cpp]: could this be a change in file encoding?

Comment (kd-11) [reply rpcs3/Emu/Cell/PPUModule.cpp]: It is, but if its now with BOM it should be ok

**Comment by kd-11** [comment]
Looks to be somewhat more complete; just needs a review from neko.

**Comment by Nekotekina** [comment]
You cannot use native wchar functions.

**Comment by RipleyTom** [comment]
I removed the wchar functions.

File: rpcs3/Emu/Cell/Modules/StaticHLE.cpp [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+		}
+		if (pattern[3].size() != 4)
+		{
+			static_hle.error("[d]:Total length != 4", i);
+			continue;
+		}
+
+		shle_pattern dapat;
+
+		auto char_to_u8 = [&](u8 char1, u8 char2) -> u16 {
```
Comment (kd-11) [inline rpcs3/Emu/Cell/Modules/StaticHLE.cpp]: brace on newline

**Pull request merged**